src/components/shared/SignupModal.tsx

```diff
@@ -0,0 +1,46 @@
+interface SignupModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+const SignupModal = ({ isOpen, onClose }: SignupModalProps) => {
+  return (
+    <div>
+      <label
+        htmlFor="my-modal-3"
+        className={`modal cursor-pointer ${isOpen && "modal-open"}`}
+      >
+        <div className="modal-box relative">
+          <label
+            htmlFor="my-modal-3"
+            onClick={() => onClose()}

```

Comment:
```suggestion
            onClick={onClose}
```
nit: you can just pass it as a callback